.github/workflows/unit_tests.yml

```
@@ -27,23 +27,30 @@ jobs:
+
+      # use github python action instead of uv to take advantage of caching
       - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python }}
-          cache: 'pip'
-          cache-dependency-path: '**/pyproject.toml'
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
+         with:
```

actionlint (1.7.4)

could not parse as YAML: yaml: line 39: mapping values are not allowed in this context

(syntax-check)